classes/class-wc-connect-service-settings-store.php

```diff
@@ -122,6 +122,71 @@ public function update_label_order_meta_data( $order_id, $new_label_data ) {
 			update_post_meta( $order_id, 'wc_connect_labels', json_encode( $labels_data ) );
 		}
 
+		private function get_user_name( $id ) {
+			$data = get_userdata( $id );
+			if ( ! $data ) {
+				return null;
+			}
+
+			return $data->user_login;
+		}
+
+		public function get_current_user_data() {
+			$user_id = get_current_user_id();
+
+			return array(
+				'user_id' => $user_id,
+				'user_name' => $this->get_user_name( $user_id ),
+				'user_url' => get_edit_user_link( $user_id ),
+			);
+		}
+
+		private function decorate_history_entry( $entry ) {
+			$user_id = $entry[ 'user_id' ];
+
+			$entry[ 'user_name' ] = $this->get_user_name( $user_id );
+			$entry[ 'user_url' ] = get_edit_user_link( $user_id );
+
+			return $entry;
+		}
+
+		public function update_label_order_history( $order_id, $label_ids, $entry_type, $time ) {
```

Comment:
All uses of this pass in `$time = time() * 1000`. Perhaps this can just be an optional argument with a default?

Additionally, what do you think about handling both single `$label_id` and an array of `$label_ids`? That way, function calls don't have to wrap `array( $a_single_label_id )`.